### PR TITLE
Add more documentation to `TimeZone`

### PR DESCRIPTION
**Pull request by pitdicker**
Split out from https://github.com/chronotope/chrono/pull/1040.

### CODE REVIEW

File: src/offset/mod.rs [hunk 1]
```diff
@@ -209,10 +209,119 @@ pub trait Offset: Sized + Clone + fmt::Debug {
...
-/// [`DateTime`](../struct.DateTime.html) types.
+/// # Using `TimeZone`
+///
+/// Most methods of this trait are deprecated, it is usually better or easier to work with the
+/// methods on [`DateTime`] and [`NaiveDateTime`] instead.
+///
+/// A type implementing [`TimeZone`] can be used to create a [`DateTime`]:
+///
+/// Creating a [`DateTime`] from a [`NaiveDateTime`]:
+/// - [`from_local_datetime`](#method.from_local_datetime) if the datetime is local.
```
Comment (jtmoon79) [inline src/offset/mod.rs:222]: ... is in local timezone.

File: src/offset/mod.rs [hunk 2]
```diff
@@ -209,10 +209,119 @@ pub trait Offset: Sized + Clone + fmt::Debug {
...
+/// - [`from_utc_datetime`](#method.from_utc_datetime) if the datetime is in utc.
```
Comment (jtmoon79) [inline src/offset/mod.rs:223]: ... is in UTC timezone.

File: src/offset/mod.rs [hunk 3]
```diff
@@ -209,10 +209,119 @@ pub trait Offset: Sized + Clone + fmt::Debug {
...
+/// be given as either local or UTC.
+///
+/// [`from_offset`](#tymethod.from_offset) allows the type to be reconstructed from the associated
+/// [`Offset`] type, which gets stored in a [`DateTime`].
+///
+/// ## Example
+///
+#[cfg_attr(not(feature = "clock"), doc = "```ignore")]
+#[cfg_attr(feature = "clock", doc = "```rust")]
+/// use chrono::{LocalResult, Offset, TimeZone};
```
Comment (jtmoon79) [inline src/offset/mod.rs:251]: Why not one `use` statement?

**Review by jtmoon79 — APPROVED** [review]
Approve with nitpick comments.

File: src/offset/mod.rs (hunk 3 above)
Comment (pitdicker) [reply src/offset/mod.rs]: I had some reasoning... But if I can't remember it must not have been that good :smile:.

**Comment by pitdicker** [comment]
I'll give this another try in the future.

**Pull request closed without merging**